enterprise/server/backends/redis_execution_collector/redis_execution_collector.go

```diff
@@ -150,3 +264,42 @@ func (c *collector) DeleteExecutions(ctx context.Context, iid string) error {
 func (c *collector) DeleteInvocationLinks(ctx context.Context, executionID string) error {
 	return c.rdb.Del(ctx, getInvocationLinkKey(executionID)).Err()
 }
+
+func (c *collector) DeleteReverseInvocationLinks(ctx context.Context, invocationID string) error {
+	return c.rdb.Del(ctx, getReverseInvocationLinkKey(invocationID)).Err()
+}
+
+func unmarshalStoredInvocationLinks(serializedResults []string) ([]*sipb.StoredInvocationLink, error) {
+	res := make([]*sipb.StoredInvocationLink, 0)
```

Comment:
nit: let's make this len(serializedResults) long and assign directly into the slice so we save allocs?

Reply:
Done (I added a slice capacity and stuck with `append`).

Recently I've been wondering whether pre-sizing on `length` vs. `capacity` is better :thinking: Pre-sizing based on length is nice because it's shorter, because you can write `slice[i] = x` instead of `slice = append(slice, x)`. But I've hit bugs before where I pre-allocate the slice based on length, then later make a change that short-circuits the loop, then wind up with some `nil` at the end of the slice, then the code panics. I feel like sticking with `append` is probably safer overall, and pre-allocating the capacity to avoid allocations (and it's probably just as efficient, if I had to guess).